client/js/main.ts

```diff
@@ -93,24 +140,35 @@ function delay (milliseconds: number) {
 	});
 }
 
-function statusFormatter (time: Date, by: string = "unknown"): string {
+function statusFormatter (time: string, by: string = "unknown"): string {
 	// Escape possible HTML in username
 	by = by.replace(/&/g, "&amp;").replace(/</g, "&lt;").replace(/>/g, "&gt;");
-
-	return `Checked in <abbr title="${moment(time).format("dddd, MMMM Do YYYY, h:mm:ss A")}">${moment(time).fromNow()}</abbr> by <code>${by}</code>`;
+	const date: Date = new Date(time);
+	return `Checked in <abbr title="${moment(date).format("dddd, MMMM Do YYYY, h:mm:ss A")}">${moment(time).fromNow()}</abbr> by <code>${by}</code>`;
 }
 
 function checkIn (e: Event) {
 	let button = (<HTMLButtonElement> e.target)!;
 	let isCheckedIn: boolean = button.classList.contains("checked-in");
 	button.disabled = true;
 	let tag: string = tagSelector.value;
-
-	qwest.post("/api/checkin", {
-		id: button.parentElement!.parentElement!.id.slice(5),
-		revert: isCheckedIn ? "true" : "false",
-		tag: tag
-	}).catch((e, xhr, response) => {
+	let id: string = button.parentElement!.parentElement!.id.slice(5);
+	let action: string = isCheckedIn ? "check_out" : "check_in";
+
+	let mutation: string = `mutation {
+	  ${action}(user: "${id}", tag: "${tag}") {
```

Comment:
We should try to avoid interpolating variables ourselves in GraphQL calls. This can be abused similarly to SQL injection. http://graphql.org/graphql-js/passing-arguments/

Reply:
This is a really good point. I will switch to using variables to pass in arguments!